python/tests/test_table_read.py

```diff
@@ -495,6 +495,14 @@ def test_writer_fails_on_protocol():
         dt.to_pandas()
 
 
+@pytest.mark.parametrize("version, expected", [(2, (5, 3))])
+def test_peek_next_commit(version, expected):
+    table_path = "../rust/tests/data/simple_table"
+    dt = DeltaTable(table_path)
+    actions, current_version = dt.peek_next_commit(version=version)
+    assert (len(actions), current_version) == expected
+
```

Comment:
maybe we could add a test to cover DeltaLogNotFound.

Reply:
Added

Reply:
I guess there's no way to test the loop? I'm thinking. `dt.peek_next_commit(version)` where `version` does not have an object but `version+1` does and `_latest_version` is currently `version`.

Reply:
I'm a little confused here, do you mean use version=-1 as input or something else?

Reply:
I was thinking there'd be a unit test in which the underlying delta table has actual latest version `10`, the `DeltaTable` thinks the latest version is `6`, and version `7` doesn't actually exist. So we call `dt.peek_next_commit(6)` and expect the next commit to be `8`, which requires

1. realizing `7` is bigger than the thought-to-be latest version `6`
2. updating the latest version which is now `10`
3. checking if version `7` is there, which it isn't
4. trying again with version `8` and finding version `8` to return.

Reply:
Added data that missed a commit and a test for it. `get_latest_version` function can get the actual latest version when there is a missed commit (e.g., if version 7 doesn't exist and the actual latest version is 10, `DeltaTable.get_latest_version` will return 10), so we don't need to update the latest version.